akka-remote/src/main/scala/akka/remote/artery/Association.scala

```diff
@@ -352,6 +356,26 @@ private[remote] class Association(
       deadletters ! env
     }
 
+    def shouldSendDeathWatchNotification(d: DeathWatchNotification): Boolean =
+      d.addressTerminated || !transport.provider.settings.HasCluster || !transport.system.isTerminating()
+
+    def sendSystemMessage(outboundEnvelope: OutboundEnvelope): Unit = {
+      outboundEnvelope.message match {
+        case d: DeathWatchNotification if !shouldSendDeathWatchNotification(d) =>
```

Comment:
Added this, so it's not sending the DeathWatchNotification if using Cluster and the actor system is terminating. The other side will be notified by AddressTerminated when member has been removed.

I actually found that sometimes it's not sending the DeathWatchNotification, also before this change. We were probably not fully aware of that but it makes me more comfortable with the change. When ClusterRemoteWatcher sees MemberRemoved it will publish AddressTerminated. So that will happen for all other members in the cluster when the ActorSystem is terminating. In DeathWatch.scala that will be received and the `watchedBy` will be removed (sometimes) before `tellWatchersWeDied`. In that case DeathWatchNotification message will not be sent to the watchers.

https://github.com/akka/akka/blob/9641780054bc587139fd348739aee750a7016941/akka-actor/src/main/scala/akka/actor/dungeon/DeathWatch.scala#L199-L203